### PR TITLE
NEW Remove Phockito test dependency, update config API use, update Travis config

### LINKED ISSUE #40: Use codecov.io for code coverage
#38 removed some old configuration from `.scrutinizer.yml` for external code coverage since Travis was not configured to provide it.

We should switch to using Codecov.io instead.

### PR DESCRIPTION
**Pull request by robbieaverill**
Also adds Codecov.io integration over Scrutinizer (fixes #40)

### CODE REVIEW

**Comment by robbieaverill** [comment]
Scrutinizer is still connected to silverstripe-labs FYI

File: .travis.yml [hunk 1]
```diff
@@ -1,24 +1,32 @@
...
-  - cd ~/builds/ss
-  - composer install
+  - phpenv rehash
+  - phpenv config-rm xdebug.ini
+
+  - composer install --prefer-dist
+  - composer require --prefer-dist --no-update silverstripe/framework:4.0.x-dev
+  - composer update
+
+  - if [[ $DB == PGSQL ]]; then composer require silverstripe/postgresql:2.0.x-dev --prefer-dist; fi
```
Comment (dhensby) [inline .travis.yml]: Move this to above the `composer update` line and add `--no-update` for faster installs

File: composer.json [hunk 2]
```diff
@@ -18,8 +18,9 @@
         "silverstripe/framework": "^4.0@dev"
     },
     "require-dev": {
-        "hafriedlander/phockito": "dev-master",
-        "silverstripe/versioned": "^1.0@dev"
+        "silverstripe/versioned": "^1.0@dev",
+        "phpunit/phpunit": "^5.7",
+        "squizlabs/php_codesniffer": "*"
```
Comment (dhensby) [inline composer.json]: add a constraint

**Comment by dhensby** [comment]
> Scrutinizer is still connected to silverstripe-labs FYI

I've updated scrutinizer

**Comment by robbieaverill** [comment]
Merging at @dhensby's approval

**Comment by robbieaverill** [comment]
Good to merge if you're happy @dhensby

**Pull request merged**